services/addons/images/count_metric/gen_email.py

```diff
@@ -62,13 +62,21 @@ def generate_count_table(rsu_dict, message_type_list):
             in_count = value["counts"][type]["in"]
             out_count = value["counts"][type]["out"]
 
-            # Normalize the diff_percent depending on message types that are deduplicated to 1/hour
-            x = 3600 if type.lower() == "map" or type.lower() == "tim" else 1
-            value["counts"][type]["diff_percent"] = (
-                abs(out_count / -(-(in_count / x) // 1) - 1) * 100
-                if in_count != 0
-                else (5 if out_count > in_count else 0)
-            )
+            if type.lower() == "bsm" or type.lower() == "tim":
+                # For unique deduplication situations, don't validate counts unless zero
+                value["counts"][type]["diff_percent"] = (
+                    6
+                    if (in_count != 0 and out_count == 0) or (out_count > in_count)
+                    else 0
+                )
+            else:
+                # Normalize the diff_percent depending on message types that are deduplicated to 1/hour
+                x = 3600 if type.lower() == "map" else 1
+                value["counts"][type]["diff_percent"] = (
+                    abs(out_count / -(-(in_count / x) // 1) - 1) * 100
+                    if in_count != 0
+                    else (6 if out_count > in_count else 0)
+                )

```

Comment:
Suggestion (non-blocking): Consider extracting this into a helper method with a clear, descriptive name. This could improve code clarity and make it easier for readers to understand.

Reply:
Added a comment to help provide further clarification.